### PR TITLE
feat: [10291] redirect to workspaces view when wrong id in url

### PR DESCRIPTION
**Pull request by esasova**
Before creating a PR, please check that:
- [ ] Code is clean
- [ ] Tests are still working (cypress tests and `yarn test`)
- [ ] Changes don't cause new errors or warnings in browser console
- [ ] Documentation is up-to-date
- [ ] Breaking changes are clearly identified with [conventional commits](https://kapeli.com/cheat_sheets/Conventional_Commits.docset/Contents/Resources/Documents/index)

### CODE REVIEW

File: public/locales/fr/translation.json [hunk 1]
```diff
@@ -465,6 +465,11 @@
             "contact": "Si vous pensez qu'il s'agit d'une erreur, veuillez contacter votre administrateur."
           }
         }
+      },
+      "error": {
+        "title": "Erreur d'initialisation de la web app",
+        "message": "Le workspace avec id {{workspaceId}} n'existe pas dans la liste des workspaces",
```
Comment (csm-thu) [inline public/locales/fr/translation.json:471]: J'aurais plutôt mis `avec l'id` ou `avec l'identifiant`

File: src/AppRoutes.js [hunk 2]
```diff
@@ -7,14 +7,14 @@ import PropTypes from 'prop-types';
...
-import { useWorkspacesList } from './state/hooks/WorkspaceHooks';
+import { useWorkspace } from './state/hooks/WorkspaceHooks';
 
 const AppRoutes = (props) => {
   const { authenticated, authorized, tabs } = props;
   const location = useLocation();
   const providedUrl = sessionStorage.getItem('providedUrl');
   const providedUrlBeforeSignIn = sessionStorage.getItem('providedUrlBeforeSignIn');
-  const workspacesList = useWorkspacesList();
+  const currentWorkspace = useWorkspace();
```
Comment (csm-thu) [inline src/AppRoutes.js:17]: If we don't use `workspace.status`, we can use the `useWorkspaceData` hook instead :slightly_smiling_face:

**Review by csm-thu — APPROVED** [review]
Minor changes suggested, but it LGTM :+1:

File: src/AppRoutes.js (hunk 2 above)
Comment (esasova) [reply src/AppRoutes.js:17]: good idea, thanks !

File: src/AppRoutes.js [hunk 3]
```diff
@@ -7,14 +7,14 @@ import PropTypes from 'prop-types';
...
-import { useWorkspacesList } from './state/hooks/WorkspaceHooks';
+import { useWorkspaceData } from './state/hooks/WorkspaceHooks';
 
 const AppRoutes = (props) => {
   const { authenticated, authorized, tabs } = props;
   const location = useLocation();
   const providedUrl = sessionStorage.getItem('providedUrl');
   const providedUrlBeforeSignIn = sessionStorage.getItem('providedUrlBeforeSignIn');
-  const workspacesList = useWorkspacesList();
+  const currentWorkspaceData = useWorkspaceData();
```
Comment (ardvi) [inline src/AppRoutes.js:17]: it's seems that you use only the id of workspace, maybe you can use the useWorkspaceId hook instead

**Review by ardvi — APPROVED** [review]
LGTM , just a remark

**Review by nborde-CSM — APPROVED** [review]
LGTM :+1:

**Pull request merged**